src/components/body/ammo-body.js

```diff
@@ -0,0 +1,480 @@
+/* global Ammo,THREE */
+const AmmoDebugDrawer = require("ammo-debug-drawer");
+const threeToAmmo = require("three-to-ammo");
+
+//Activation States
+const ACTIVE_TAG = 1;
+const ISLAND_SLEEPING = 2;
+const WANTS_DEACTIVATION = 3;
+const DISABLE_DEACTIVATION = 4;
+const DISABLE_SIMULATION = 5;
+
+//CollisionFlags
+const CF_STATIC_OBJECT = 1;
+const CF_KINEMATIC_OBJECT = 2;
+const CF_NO_CONTACT_RESPONSE = 4;
+const CF_CUSTOM_MATERIAL_CALLBACK = 8; //this allows per-triangle material (friction/restitution)
+const CF_CHARACTER_OBJECT = 16;
+const CF_DISABLE_VISUALIZE_OBJECT = 32; //disable debug drawing
+const CF_DISABLE_SPU_COLLISION_PROCESSING = 64; //disable parallel/SPU processing
+
+function almostEqualsVector3(epsilon, u, v) {
+  return Math.abs(u.x - v.x) < epsilon && Math.abs(u.y - v.y) < epsilon && Math.abs(u.z - v.z) < epsilon;
+}
+
+function almostEqualsQuaternion(epsilon, u, v) {
+  return (
+    (Math.abs(u.x - v.x) < epsilon &&
+      Math.abs(u.y - v.y) < epsilon &&
+      Math.abs(u.z - v.z) < epsilon &&
+      Math.abs(u.w - v.w) < epsilon) ||
+    (Math.abs(u.x + v.x) < epsilon &&
+      Math.abs(u.y + v.y) < epsilon &&
+      Math.abs(u.z + v.z) < epsilon &&
+      Math.abs(u.w + v.w) < epsilon)
+  );
+}
+
+let AmmoBody = {
+  schema: {
+    loadedEvent: { default: "" },
+    mass: { default: 1 },
+    gravity: { type: "vec3", default: { x: 0, y: -9.8, z: 0 } },
+    linearDamping: { default: 0.01 },
+    angularDamping: { default: 0.01 },
+    linearSleepingThreshold: { default: 1.6 },
+    angularSleepingThreshold: { default: 2.5 },
+    angularFactor: { type: "vec3", default: { x: 1, y: 1, z: 1 } },
+    activationState: {
+      default: ACTIVE_TAG,
+      oneOf: [ACTIVE_TAG, ISLAND_SLEEPING, WANTS_DEACTIVATION, DISABLE_DEACTIVATION, DISABLE_SIMULATION]
+    },
+    type: { default: "dynamic", oneOf: ["static", "dynamic", "kinematic"] },
+    addCollideEventListener: { default: false },
+    collisionFlags: { default: 0 }, //32-bit mask
+    collisionFilterGroup: { default: 1 }, //32-bit mask,
+    collisionFilterMask: { default: 1 }, //32-bit mask
```

Comment:
Does A-Frame parse hexadecimal inputs correctly? That would be ideal, if so.

Reply:
I'm not sure hexadecimal is going to be any clearer. Maybe if A-Frame supported binary that would be better? I don't think it does though.

Reply:
For example, I find this easier:

```js
const group = 0x0ff0;
const a = 0xf000; // group & a == false
const b = 0x0f00; // group & b == true
const c = 0x00f0; // group & c == true
const d = 0x000f; // group & d == false
```

But i haven't checked whether A-Frame supports it, we can clean up the docs later if so.